### PR TITLE
Cleanup of USB peripheral driver

### PR DESCRIPTION
**Pull request by phil-levis-google**
This is a partial code cleanup of the USB peripheral driver. It changes all of the registers that aren't direct integer values (i.e., have fields) to use the Register create. It also removes the remaining few magic constants inherited from the original code base and replaces them with properly formatted register fields/values.

I tested using U2FTest/u2f_app that the peripheral driver is still working properly.

```
----------------------
`make prtest` summary:
----------------------
git rev-parse HEAD
5b51cd0e78d4b4fae8a87f043c6ae85e2002bbce
git status
On branch usb_cleanup
Your branch is up to date with 'origin/usb_cleanup'.

nothing to commit, working tree clean
```

### CODE REVIEW

File: h1b/src/usb/mod.rs [hunk 1]
```diff
@@ -23,17 +23,30 @@ pub mod u2f;
...
+pub use self::registers::DeviceConfig;
+pub use self::registers::DeviceControl;
 pub use self::registers::DMADescriptor;
+pub use self::registers::EndpointControl;
+pub use self::registers::Gpio;
+pub use self::registers::InEndpointInterruptMask;
+pub use self::registers::Interrupt;
+pub use self::registers::OutEndpointInterruptMask;
+pub use self::registers::Reset;
+pub use self::registers::UsbConfiguration;
```
Comment (jrvanwhy) [inline h1b/src/usb/mod.rs]: It may be worth doing `pub use self::registers::{AhbConfig, AllEndpointInterrupt, Descriptor, ...};` here.

File: h1b/src/usb/registers.rs [hunk 2]
```diff
@@ -183,6 +458,9 @@ impl BitAnd for EpCtl {
     }
 }
 
+
+
+
```
Comment (jrvanwhy) [inline h1b/src/usb/registers.rs]: Is this change intentional? 5 newlines in a row is a lot.

**Review by jrvanwhy — APPROVED** [review]

File: h1b/src/usb/mod.rs (hunk 1 above)
Comment (phil-levis-google) [reply h1b/src/usb/mod.rs]: Good point -- I've reformatted all of the use statements to be cleaner.

File: h1b/src/usb/registers.rs (hunk 2 above)
Comment (phil-levis-google) [reply h1b/src/usb/registers.rs]: Unintentional. Fixed.
